lisp/ess-utils.el

```diff
@@ -92,12 +92,61 @@ Used in noweb modes.")
 
 ;;*;; elisp tools
 
+(defun ess-next-code-line (&optional arg skip-to-eob)
```

Comment:
Should we namespace these functions in `ess--`? It feels like we should.

Reply:
> It feels like we should.

Why?

Reply:
The question should always go in the opposite direction: why should be export this function? Exporting makes it harder to refactor and change things later.

Reply:
> The question should always go in the opposite direction:

I'm not sure I agree. If we've written useful code, others should be able to use it too if they want.

Reply:
They should, but if we haven't exported the function, they should also be prepared to change their code on update.

I think these functions need some more consideration especially regarding ess-r-syntax and the likely changes coming for this release. We don't have to make them public right away.

Reply:
> but if we haven't exported the function, they should also be prepared to change their code on update.

they're already non-interactive, so I'm guessing not very many people will make direct use of them.

> these functions need some more consideration

Well, you can always say that :-) 

It's worth pointing out that ess-next-code-line isn't a new function in this PR. It just got moved to ess-utils.

Reply:
Oh good, then forget about it.

Most syntax functions are namespaced in `ess-` instead of `ess-r--` because I didn't know better at the time. I'm going to clean that up. I think generic syntax stuff would be very hard to achieve so has to be language-specific.

Reply:
(but the next code line stuff does look generic)